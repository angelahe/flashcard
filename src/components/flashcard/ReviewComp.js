import React from 'react';
import PropTypes from 'prop-types';
import '../../styles/flashcard.css';

class ReviewComp extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentCard: '',
      deckList: [],
    };
  }

  handleDeckDone = () => {
    const { onDeckDone } = this.props;
    onDeckDone();
  }

  render() {
    const { deck } = this.props;
    return (
      <div className="ReviewComp">
        <p>Review Comp</p>
        <p>{deck}</p>
      </div>
    );
  }
}

ReviewComp.defaultProps = {
  onDeckDone: () => { },
};

ReviewComp.propTypes = {
  deck: PropTypes.string.isRequired,
  onDeckDone: PropTypes.func,
};

export default ReviewComp;
