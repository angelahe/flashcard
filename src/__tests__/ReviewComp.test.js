import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import ReviewComp from '../components/flashcard/ReviewComp';

it('renders without crashing', () => {
  shallow(<ReviewComp />);
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ReviewComp />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('snapshot of UI renders consistently', () => {
  const tree = renderer.create(<ReviewComp />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('contains the container div ChooseLevelComp on shallow render', () => {
  const component = shallow(<ReviewComp />);
  expect(component.html()).toContain('ReviewComp');
});
