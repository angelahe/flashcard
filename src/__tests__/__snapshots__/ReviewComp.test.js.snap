// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of UI renders consistently 1`] = `
<div
  className="ReviewComp"
>
  <p>
    Review Comp
  </p>
  <p />
</div>
`;
